### PR TITLE
CLN: annotations in core.dtypes

### PR DESCRIPTION
**Pull request by jbrockmendel**
Some docstring cleanup.

### CODE REVIEW

File: pandas/core/dtypes/base.py [hunk 1]
```diff
@@ -233,16 +235,19 @@ def construct_from_string(cls, string: str):
         ...                         "'{}'".format(cls.__name__, string))
         """
         if not isinstance(string, str):
-            raise TypeError("Expects a string, got {}".format(type(string)))
+            raise TypeError("Expects a string, got {typ}".format(typ=type(string)))
```
Comment (simonjayhawkins) [inline pandas/core/dtypes/base.py]: if changing .format could now use f-strings?

File: pandas/core/dtypes/base.py [hunk 2]
```diff
@@ -233,16 +235,19 @@ def construct_from_string(cls, string: str):
...
         if string != cls.name:
             raise TypeError(
-                "Cannot construct a '{}' from '{}'".format(cls.__name__, string)
+                "Cannot construct a '{cls}' from '{string}'".format(
```
Comment (simonjayhawkins) [inline pandas/core/dtypes/base.py]: same

File: pandas/core/dtypes/dtypes.py [hunk 3]
```diff
@@ -416,12 +416,12 @@ def __eq__(self, other: Any) -> bool:
...
 
     def __repr__(self) -> str_type:
-        tpl = "CategoricalDtype(categories={}ordered={})"
+        tpl = "CategoricalDtype(categories={data}ordered={ordered})"
         if self.categories is None:
             data = "None, "
         else:
             data = self.categories._format_data(name=self.__class__.__name__)
-        return tpl.format(data, self._ordered)
+        return tpl.format(data=data, ordered=self._ordered)
```
Comment (simonjayhawkins) [inline pandas/core/dtypes/dtypes.py]: same

File: pandas/core/dtypes/dtypes.py [hunk 4]
```diff
@@ -871,7 +871,7 @@ def _parse_dtype_strict(cls, freq):
         raise ValueError("could not construct PeriodDtype")
 
     @classmethod
-    def construct_from_string(cls, string):
+    def construct_from_string(cls, string: str_type):
```
Comment (simonjayhawkins) [inline pandas/core/dtypes/dtypes.py]: the code looks like it accepts DateOffset

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @jbrockmendel. f-string suggestion and minor query otherwise lgtm.

File: pandas/core/dtypes/base.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/dtypes/base.py]: oh golly, im still tracking down %-based usage

Comment (alimcmaster1) [reply pandas/core/dtypes/base.py]: I've had a go at upgrading some of our % usage - hopefully there isn't too much more!

https://github.com/pandas-dev/pandas/pull/29518/files

File: pandas/core/dtypes/dtypes.py (hunk 4 above)
Comment (jbrockmendel) [reply pandas/core/dtypes/dtypes.py]: Good point.  Maybe it shouldn't?

Comment (jbrockmendel) [reply pandas/core/dtypes/dtypes.py]: reverted this change, punting on this for now

**Comment by jreback** [comment]
can you rebase

**Comment by jbrockmendel** [comment]
rebased+green

**Comment by jreback** [comment]
thanks

**Pull request merged**